lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -376,6 +376,51 @@ class DecomposeAtenAddmmOp : public OpRewritePattern<AtenAddmmOp> {
 };
 } // namespace
 
+// Decompose torch.mean into: sum(x)/div(numTensorElements).
+namespace {
+class DecomposeAtenMeanOp : public OpRewritePattern<AtenMeanOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMeanOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.self();
+    Value output = op.result();
+
+    Value dtype = rewriter.create<ConstantNoneOp>(loc);
+    Value sum = rewriter.create<AtenSumOp>(loc, output.getType(), input, dtype);
+    if (!sum)
+      return failure();
+
+    BaseTensorType tensorType = input.getType().cast<BaseTensorType>();
+    SmallVector<int64_t> sizes;
+    int64_t tensorSize = 1;
+    if (tensorType.hasSizes()) {
+      ArrayRef<int64_t> inputShape = tensorType.getSizes();
+      for (int64_t i : inputShape) {
+        if (i < 1)
+          return rewriter.notifyMatchFailure(op, "Invalid tensor size.");
+        tensorSize *= i;
```

Comment:
I don't think this works for dynamic dimensions. You should be able to calculate the size of each dimension using `AtenSizeIntOp` which takes a tensor and a dimension and gets the size of that dimension.

Another option could be to add a lowering for `torch.numel` and use that to get the total number of elements.